### PR TITLE
Add missing resources to ControlStrings.resx, use nameof for resource lookups

### LINKED ISSUE #1440: Attempting to view some metrics fails
For some reason, commit fafff275f29aa02aec4cfe413d9fa50fbea161f3 caused exceptions to be raised when viewing some metrics (including process.runtime.dotnet.gc.collections.count). 

Filed for tracking.

### PR DESCRIPTION
**Pull request by adamint**
Fixes #1440

Three strings were erroneously removed from `ControlsStrings.resx`, and were added back; the generated c# files should have then removed the respective fields, which would have caught this, but they did not.

Additionally, we should be using nameof instead of directly referencing the field values in localization lookups, so I made those changes project-wide.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1445)

### CODE REVIEW

File: src/Aspire.Dashboard/Resources/ControlsStrings.Designer.cs [hunk 1]
```diff
@@ -141,7 +141,7 @@ public class ControlsStrings {
         }
         
         /// <summary>
-        ///   Looks up a localized string similar to Show Spec Only.
+        ///   Looks up a localized string similar to .
```
Comment (drewnoakes) [inline src/Aspire.Dashboard/Resources/ControlsStrings.Designer.cs:144]: This looks strange. Why is the documented string empty?

File: src/Aspire.Dashboard/Components/Controls/ChartContainer.razor [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 @if (_instrument is null)
 {
-    <div>@Loc[nameof(ControlsStrings.ChartContainerUnableToDisplay)]</div>
+    <div>@Loc[ControlsStrings.ChartContainerUnableToDisplay]</div>
```
Comment (drewnoakes) [inline src/Aspire.Dashboard/Components/Controls/ChartContainer.razor]: Why can't these just be:

```
<div>@ControlsStrings.ChartContainerUnableToDisplay</div>
```

**Review by drewnoakes — APPROVED** [review]

File: src/Aspire.Dashboard/Resources/ControlsStrings.resx [hunk 3]
```diff
@@ -128,7 +128,7 @@
   </data>
   <data name="EnvironmentVariablesFilterToggleShowSpecOnly" xml:space="preserve">
     <value>Show spec only</value>
-    <note>Spec means from the resource specification</note>
+    <comment>Spec means from the resource specification</comment>
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Resources/ControlsStrings.resx:131]: I fixed these early last week to be `<comment>` and then [they changed back to note](https://github.com/dotnet/aspire/pull/1332/files#diff-2453339c9205116de2ce41f77436813b3bbd248dd69d23ec85feaae87e55e23fR131).

What is changing them?

Comment (adamint) [reply src/Aspire.Dashboard/Resources/ControlsStrings.resx:131]: @JamesNK I'm guessing this was due a botched merge commit by me, considering it just happened to this one file

**Pull request merged**